### ISSUE TITLE
[Search plugin] Jumping to matches occurs in the wrong order when the same page contains different keyword matches

### ISSUE THREAD

**Issue by jgredecka** [react-pdf-viewer/react-pdf-viewer#1755]
## Context

When different keyword matches are found and highlighted on the same page, jumping to matches occurs in the wrong order, and jumping to the next match sometimes jumps back to the first match on the page. 

Some matches are also completely skipped.

## Example code
A basic sandbox reproducing this issue is [here](https://codesandbox.io/p/sandbox/react-pdf-viewer-jump-to-match-fn26tc?file=%2Fsrc%2FApp.tsx%3A7%2C32)

To reproduce:
- Click on `Click to highlight`
- Click on `Next match` 3 times
- When attempting to jump to the 4th match you'll see that it jumps back to the first match on the page instead

**Comment by isimisi** [comment]
I don't think this repo is getting maintained anymore, but [this PR I created should fix just that](https://github.com/react-pdf-viewer/react-pdf-viewer/pull/1625)
